src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,224 @@
+    .map((c) => c.trim())
+    .filter(hasCookieName) // Find the key/value pair with _csrf as key
+    .map((cookie) => cookie.substring(cookieName.length + 1)) // Retrieve value for '_csrf'-key
+
+  if (cookieValues.length == 0 || cookieValues[0] === '') {
+    throw new GiteaDroneError(`No value for ${cookieName} was found`)
+  }
+  return cookieValues[0]
+}
+
+async function authorizeOAuthApp(giteaUrl: string, droneLoginUrl: string, oauthData: DroneSecret) {
+  let authorizeHeaderCookies: string[]
+  try {
+    authorizeHeaderCookies = await getGiteaAuthorizationHeaderCookies(giteaUrl, droneLoginUrl, oauthData)
+  } catch (error) {
+    throw new GiteaDroneError('Authorization already granted or something went wrong')
+  }
+
+  const csrfToken = getCookieValue(authorizeHeaderCookies, csrfCookieName)
+
+  console.log('Granting authorization')
+
+  const grantOptions: AxiosRequestConfig = {
+    method: 'POST',
+    url: `${giteaUrl}/login/oauth/grant`,
+    headers: {
+      cookie: authorizeHeaderCookies.map((cookie) => cookie.split(';')[0]).join('; '),
+      Cookie: authorizeHeaderCookies.map((cookie) => cookie.split(';')[0]).join('; '),
+    },
+    maxRedirects: 1,
+    auth: {
+      username: env.GITEA_USER,
+      password: env.GITEA_PASSWORD,
+    },
+    // Data for this post query must be stringified https://github.com/axios/axios#using-applicationx-www-form-urlencoded-format
+    data: querystring.stringify({
+      _csrf: csrfToken,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      client_id: `${oauthData.clientId}`,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      redirect_uri: droneLoginUrl,
+    }),
+  }
+
+  try {
+    await axios.request(grantOptions)
+  } catch (error) {
+    // Do nothing, error code could be on the redirect
+  }
+}
+
+export function isSecretValid(remoteSecret: DroneSecret, oauthApps: OAuth2Application[]): boolean {
+  if (!remoteSecret) {
+    console.log('Remote secret was not found')
+  } else if (remoteSecret.clientId === '' || remoteSecret.clientSecret === '') {
+    console.log('Remote secret values were empty')
+  } else if (oauthApps.length == 0) {
+    console.log("Gitea doesn't have any oauth application defined")
+  } else if (!oauthApps.some((e) => e.clientId === Buffer.from(remoteSecret.clientId, 'base64').toString())) {
+    console.log('OAuth data did not match with expected secret')
+  } else {
+    return true
+  }
+  return false
+}
+async function createK8SSecret(apiClient: k8s.CoreV1Api, result: DroneSecret) {
+  const secret: k8s.V1Secret = new k8s.V1Secret()
+  secret.metadata = new k8s.V1ObjectMeta()
+  secret.metadata.name = k8sSecretName
+  secret.metadata.namespace = k8sNamespace
+  secret.data = {
+    clientId: Buffer.from(result.clientId).toString('base64'),
+    clientSecret: Buffer.from(result.clientSecret).toString('base64'),
+  }
```

Comment:
Please use shorthand notation as you found in example api code:
```suggestion
  const secret = {
    ...new k8s.V1Secret(),
    metadata: { ...new k8s.V1ObjectMeta(), name: k8sSecretName },
    type: 'docker-registry',
    data: {
      clientId: Buffer.from(result.clientId).toString('base64'),
      clientSecret: Buffer.from(result.clientSecret).toString('base64'),
    }
  }
```